main.go

```diff
@@ -0,0 +1,139 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os"
+	"sort"
+	"strings"
+	"sync"
+)
+
+type Command struct {
+	Name     string
+	Argument string
+}
+
+func readInput() (<-chan Command, []string) {
+	cmds := stdIn()
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	var filePath string
+	out := make(chan Command, len(cmds))
+
+	go func(cmds []string, filePath *string) {
+		for i, cmd := range cmds {
+			sepCmd := Split(cmd)
+			if i == 0 && len(sepCmd) == 1 {
```

Comment:
i==0 is not obvious for other developers what will happen here. As I understand you check is filepath present here. You can replace it with `if cmd == cat && len(sepCmd) == 1 {log.Fatal}